libfreerdp/codec/dsp.c

```diff
@@ -111,6 +111,90 @@ struct _FREERDP_DSP_CONTEXT
 #endif
 };
 
+static BOOL freerdp_dsp_channel_mix(FREERDP_DSP_CONTEXT* context,
+                                    const BYTE* src, size_t size,
+                                    const AUDIO_FORMAT* srcFormat,
+                                    const BYTE** data, size_t* length)
+{
+	UINT32 bpp;
+	size_t samples;
+	size_t x, y;
+
+	if (!context || !data || !length)
+		return FALSE;
+
+	if (srcFormat->wFormatTag != WAVE_FORMAT_PCM)
+		return FALSE;
+
+	bpp = srcFormat->wBitsPerSample > 8 ? 2 : 1;
```

Comment:
This will cause problems with formats that use less than 8 bits per sample (i.e. ADPCM)